chain-config/tests/chain_config_blackbox_tests.rs

```diff
@@ -289,3 +294,258 @@ fn test_update_config() {
             assert!(config.min_validators == 1 && config.max_validators == 2);
         });
 }
+
+/// ### TEST
+/// C-CONFIG_REGISTER_VALIDATOR_FAIL
+///
+/// ### ACTION
+/// Call 'register()' during the setup phase
+///
+/// ### EXPECTED
+/// Error SETUP_PHASE_NOT_COMPLETED
+#[test]
+fn test_register_validator_setup_not_completed() {
+    let mut state = ChainConfigTestState::new();
+
+    state
+        .common_setup
+        .deploy_chain_config(OptionalValue::None, None);
+
+    let new_validator = ValidatorInfo {
+        address: USER_ADDRESS.to_managed_address(),
+        bls_key: ManagedBuffer::from("validator1"),
+        egld_stake: BigUint::default(),
+        token_stake: EsdtTokenData::default(),
+    };
+
+    state.register(&new_validator, Some(SETUP_PHASE_NOT_COMPLETED), None);
+}
+
+/// ### TEST
+/// C-CONFIG_REGISTER_VALIDATOR_FAIL
+///
+/// ### ACTION
+/// Call 'register()' with too many validators
+///
+/// ### EXPECTED
+/// Error VALIDATOR_RANGE_EXCEEDED
+#[test]
+fn test_register_validator_range_exceeded_too_many_validators() {
+    let mut state = ChainConfigTestState::new();
+
+    state
+        .common_setup
+        .deploy_chain_config(OptionalValue::None, None);
```

Comment:
Please use here `default_config` to be more clear that max number of validators is 1